csm_web/frontend/src/components/MentorSection.js

```diff
@@ -296,10 +297,12 @@ function zeroPadTwoDigit(num) {
 class SpacetimeEditModal extends React.Component {
   constructor(props) {
     super(props);
+    // Time string comes as HH:MM:ss, TimeInput expects HH:MM
+    let timeString = props.defaultSpacetime.startTime;
     this.state = {
-      location: "",
-      day: "",
-      time: "",
```

Comment:
I would just add a comment here for future maintainers that this is a React anti-pattern in general because if props updated the changes would not be reflected in this modal, but that it's okay in this instance because the props won't change underneath us.